webapp/src/app.tsx

```diff
@@ -1,16 +1,81 @@
 import React, {useEffect} from 'react';
+import {useDispatch} from 'react-redux';
 
 import usePluginApi from 'src/hooks/usePluginApi';
 
 import Constants from 'src/plugin_constants';
 
+import {getGlobalModalState} from './selectors';
+import {setCurrentModalState} from './reducers/currentModal';
+import {resetGlobalModalState} from './reducers/globalModal';
+
 const GetConfig = (): JSX.Element => {
-    const {makeApiRequest} = usePluginApi();
+    const {makeApiRequest, pluginState, getApiState} = usePluginApi();
+    const {modalId, data} = getGlobalModalState(pluginState);
+    const dispatch = useDispatch();
+
+    const getConnectedUserState = () => {
+        const {isLoading, data: userData} = getApiState(Constants.pluginApiServiceConfigs.getConnectedUser.apiServiceName);
+        return {isLoading, data: userData as ConnectedState};
+    };
+
+    const getSubscriptionsConfiguredState = () => {
+        const {isLoading, isSuccess} = getApiState(Constants.pluginApiServiceConfigs.checkSubscriptionsConfigured.apiServiceName);
+        return {isLoading, isSuccess};
+    };
 
     useEffect(() => {
         makeApiRequest(Constants.pluginApiServiceConfigs.getConfig.apiServiceName);
     }, []);
 
+    useEffect(() => {
+        if (modalId) {
+            switch (modalId) {
+            case 'shareRecord':
```

Comment:
If we have modalId = 'shareRecord' then we are doing nothing? Can you please elaborate a bit?

Reply:
it's a way to use switch statements for multiple cases with the same functionality. you can read more about it here https://stackoverflow.com/questions/13207927/switch-statement-for-multiple-cases-in-javascript